### DISCUSSION TITLE
hooks on relation manager is not called

### DISCUSSION THREAD

**Discussion by rabol** [filamentphp/filament#521] [Help]
Hi

I have added a relattion manager to a resource, created the form

and added 

```
protected function beforeCreate()
{
}
```

but it does not get called.

by digging into the code, it looks like the beforeCreate hook is called in CreateRecord and not in my RelationManager object.

Does that mean that I have to create a 'custom' CreateRecord class to make hooks work on creation of related records ?

**Comment by MrSharpp** [comment]
even handleRecordCreation, mutateFormDataBeforeCreate are not being called
